cocos/core/event/callbacks-invoker.ts

```diff
@@ -319,11 +351,13 @@ export class CallbacksInvoker {
                     if (info.once) {
                         this.off(key, callback, target);
                     }
-                    if (target) {
-                        callback.call(target, arg0, arg1, arg2, arg3, arg4);
+                    // Lazy check validity of callback target, 
+                    // if target is CCObject and is no longer valid, then remove the callback info directly
+                    if (!info.check()) {
+                        this.off(key, callback, target);
```

Comment:
This still doesn't solve the problem of the event listener not being removed from the EventManager after the once event off.

Reply:
I see the problem, indeed it doesn't solve the EventListener removal, I updated Node.destroy process to fix it, please review